Packs/Base/Scripts/DBotFindSimilarIncidentsByIndicators/DBotFindSimilarIncidentsByIndicators.py

```diff
@@ -0,0 +1,507 @@
+    if not incident_id:
+        incident = demisto.incidents()[0]
+        incident_id = incident['id']
+
+    # load the related indicators to the incident
+    indicators = get_all_indicators_for_incident(incident_id)
+    indicators_map = get_indicators_map(indicators)
+    indicators = list(filter(lambda x: get_number_of_invs_for_indicators(x) < max_indicators, indicators))
+    if indicators_types:
+        indicators = [x for x in indicators if x.get('indicator_type') in indicators_types]
+    if len(indicators) < limit_nb_of_indicators:
+        return_no_mututal_indicators_found_entry()
+        return_no_similar_incident_found_entry()
+        return
+    if debug == 'True':
+        demisto.results("Indicators found for the incident")
+        demisto.results(json.dumps([x.get('id') for x in indicators]))
+
+    # Get the Investigation IDs related to the indicators if the incidents
+    incident_ids = [indicator.get('investigationIDs', None) for indicator in indicators if
+                    indicator.get('investigationIDs', None)]
+    incident_ids = flatten(incident_ids)
+    p = re.compile('[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}')
+    incident_ids = [x for x in incident_ids if not p.match(x)]
+    if not incident_ids:
+        return_no_mututal_indicators_found_entry()
+        return_no_similar_incident_found_entry()
+        return
+    return_indicator_entry(incident_ids, indicators_types, indicators)
+
+    # Get indicators related to those investigations ids
+    indicators_related = get_indicators_from_incident_ids(incident_ids)
+    if indicators_types:
+        indicators_related = [x for x in indicators_related if x.get('indicator_type') in indicators_types]
+        if not indicators_related:
+            return_no_similar_incident_found_entry()
+            return
+    incidents_with_indicators = match_indicators_incident(indicators_related, incident_ids)
+    incidents_with_indicators_join = {k: join(v) for k, v in incidents_with_indicators.items()}
+    incidents_with_indicators_join.pop(incident_id, None)
+    if not bool(incidents_with_indicators_join):
+        return_no_similar_incident_found_entry()
+        return
+
+    incidents_df = pd.DataFrame.from_dict(incidents_with_indicators_join, orient='index')
+    incidents_df.columns = ['indicators']
+
+    indicators_for_incident = [' '.join(set([x.get('id') for x in indicators]))]  # type: ignore
+    incident_df = pd.DataFrame(indicators_for_incident, columns=['indicators'])
+    if debug == 'True':
+        incidents_df_debug = incidents_df.reset_index()
+        return_outputs(
+            readable_output=tableToMarkdown("Incident dataframe debug", incidents_df_debug.to_dict(orient='records'),
+                                            list(incidents_df_debug.columns)))
+
+    # Prediction
+    model = Model(p_transformation=TRANSFORMATION)
+    model.init_prediction(incident_df, incidents_df, ['indicators'])
```

Comment:
you used "init_prediction" twice for the 2 arguments, it's fine?

Reply:
they are not the same. One is without "s"

Reply:
@yguezpa it's too similar - change the name to "current_incident_df"